simulator/mips/mips_register/mips_register.h

```diff
@@ -41,7 +36,29 @@ class MIPSRegister {
     bool is_zero()       const { return value == MIPS_REG_zero; }
     bool is_mips_hi()    const { return value == MIPS_REG_hi; }
     bool is_mips_lo()    const { return value == MIPS_REG_lo; }
-    size_t to_size_t()   const { return value; }
+
+    static MIPSRegister from_cpu_index( uint8 id) {
+        assert( id < 32u);
+        return MIPSRegister( static_cast<RegNum>( id));
+    }
+    static MIPSRegister from_cp0_index( uint8 id) {
+        /* To be implemented */
+        (void) id; return MIPSRegister::zero;
```

Comment:
It's better to put `assert( false)`

Reply:
GDB doesn't know if we have implemented CP0 registers, so I think we don't want to crash simulator if it tries to read SR.

Reply:
Silent return of zero is not much better, may be even worse. I'd throw an exception in `from_gdb_index`.